app/facades/nav_facade.rb

```diff
@@ -0,0 +1,26 @@
+class NavFacade < NavService
+
+  def self.get_directions(start_address, end_address)
+    route = route_directions(start_address, end_address)[:route][:legs][0][:maneuvers]
+
+    all_directions = [] 
+    turn_distances = []
+
+    route.each do |step|
+      all_directions << step[:narrative]
+      turn_distances << step[:distance]
+    end
+    all_directions.zip(turn_distances)
```

Comment:
So glad this ended up working! love love love zip

Reply:
Yup, it's a life savor sometimes!